hyp3_gamma/dem.py

```diff
@@ -1,98 +1,37 @@
 import json
-from collections.abc import Generator
 from pathlib import Path
 from subprocess import PIPE, run
-from tempfile import TemporaryDirectory
 
-from hyp3lib import DemError
-from osgeo import gdal, ogr
+from hyp3lib import dem
+from osgeo import ogr
 
-from hyp3_gamma.util import GDALConfigManager
 
-
-DEM_GEOJSON = '/vsicurl/https://asf-dem-west.s3.amazonaws.com/v2/cop30-2021-with-cop90-us-west-2-mirror.geojson'
-
-gdal.UseExceptions()
-ogr.UseExceptions()
+def crosses_antimeridian(geometry: dict) -> bool:
+    if geometry['type'] != 'Polygon':
+        raise ValueError(f'Geometry type {geometry["type"]} is invalid; only Polygon is supported.')
+    longitudes = [point[0] for point in geometry['coordinates'][0]]
+    return any(lon < -160 for lon in longitudes) and any(160 < lon for lon in longitudes)
 
 
 def get_geometry_from_kml(kml_file: str) -> ogr.Geometry:
-    cmd = [
-        'ogr2ogr',
-        '-wrapdateline',
-        '-datelineoffset',
-        '20',
-        '-f',
-        'GeoJSON',
-        '-mapfieldtype',
-        'DateTime=String',
-        '/vsistdout',
-        kml_file,
-    ]
+    cmd = ['ogr2ogr', '-f', 'GeoJSON', '-mapfieldtype', 'DateTime=String', '/vsistdout', kml_file]
     geojson_str = run(cmd, stdout=PIPE, check=True).stdout
-    geometry = json.loads(geojson_str)['features'][0]['geometry']
+    geojson = json.loads(geojson_str)
+    geometry = geojson['features'][0]['geometry']
+    if crosses_antimeridian(geometry):
+        for point in geometry['coordinates'][0]:
+            if point[0] < 0:
+                point[0] += 360
     return ogr.CreateGeometryFromJson(json.dumps(geometry))
 
 
-def get_dem_features() -> Generator[ogr.Feature, None, None]:
-    ds = ogr.Open(DEM_GEOJSON)
-    layer = ds.GetLayer()
-    for feature in layer:
-        yield feature
-    del ds
-
-
-def intersects_dem(geometry: ogr.Geometry) -> bool:
-    for feature in get_dem_features():
-        if feature.GetGeometryRef().Intersects(geometry):
-            return True
-    return False
-
-
-def get_dem_file_paths(geometry: ogr.Geometry) -> list[str]:
-    file_paths = []
-    for feature in get_dem_features():
-        if feature.GetGeometryRef().Intersects(geometry):
-            file_paths.append(feature.GetField('file_path'))
-    return file_paths
-
-
 def utm_from_lon_lat(lon: float, lat: float) -> int:
```

Comment:
Eventually I think this function would be good to move this to hyp3-lib because I think more than one plugin has functionality like this. I'm not advocating we do it now though.